python/recipe.cpp

```diff
@@ -112,6 +118,32 @@ struct cell_connection_shim {
     }
 };
 
+std::vector<arb::cell_connection> py_recipe_shim::connections_on(arb::cell_gid_type gid) const {
+    using pybind11::isinstance;
+    using pybind11::cast;
+
+    // Aquire the GIL because it must be held when calling isinstance and cast.
+    auto guard = pybind11::gil_scoped_acquire();
+
+    // TODO: acquire gil
+    auto pycons = impl_->connections_on(gid);
+    std::vector<arb::cell_connection> cons;
+    cons.reserve(pycons.size());
+    for (unsigned i=0; i<pycons.size(); ++i) {
+        const auto& c = pycons[i];
+        if (isinstance<cell_connection_shim>(c)) {
+            cons.push_back(cast<cell_connection_shim>(c));
+        }
+        else {
+            throw std::runtime_error(
```

Comment:
Why not use `pyarb_error` as in `event_generators`?

Reply:
Fixed.